### PR TITLE
Add possibility to render submenu in MenuItem

### PR DESCRIPTION
**Pull request by sophieH29**
# MenuItem
This PR introduces possibility to add vertical submenu rendered in MenuItem.

### TODO

- [X] Conformance test
- [X] Minimal doc site example
- [ ] Stardust base theme
- [ ] Teams Light theme
- [ ] Teams Dark theme
- [ ] Teams Contrast theme
- [ ] Confirm RTL usage
- [ ] [W3 accessibility](https://www.w3.org/standards/webdesign/accessibility) check
- [ ] [Stardust accessibility](https://github.com/stardust-ui/accessibility) check
- [ ] Update glossary props table
- [ ] Update the CHANGELOG.md

# API Proposal

## Submenu

submenu

```js
const submenuData = {
     items: [
           { key: 'new', content: 'New' },
           { key: 'open', content: 'Open' },
     ]
}
```

```jsx
<MenuItem submenu={submenuData} content="File" />
```

```html
<li>
  <a> File </a>
  <ul>
    <li> <a> New </a> </a>
    <li> <a> Open </a> </a>
 </ul>
</li>
```

### CODE REVIEW

**Comment by sophieH29** [comment]
#### Having a concern regarding style:
Menu and menu items are having a transparent background. So when a submenu is rendered, it can look weird, as everything behind it is visible. 

Looking forward to your thoughts and ideas!

File: src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -31,6 +41,9 @@ class MenuItem extends UIComponent<any, any> {
     /** Shorthand for primary content. */
     content: customPropTypes.contentShorthand,
 
+    /** Initial submenuOpened value. */
+    defaultSubmenuOpened: PropTypes.bool,
+
```
Comment (mnajdova) [inline src/components/Menu/MenuItem.tsx]: This comment will be used in the docs page for describing the property. Please use more descriptive comment avoiding any properties.

Comment (sophieH29) [reply src/components/Menu/MenuItem.tsx]: @mnajdova that makes sense. But I've made the same description as for ```defaultActiveindex``` prop in Accordion and Menu components to keep consistency. Should I change it?

Comment (mnajdova) [reply src/components/Menu/MenuItem.tsx]: Aah, alright then, leave it as is.

**Comment by notandrew** [comment]
I'd expect the submenu to have the same color as the menu bar. Is it possible to apply this style?

File: docs/src/examples/components/Menu/Types/MenuExampleWithSubmenu.shorthand.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const items = [
+  { key: 'file', content: <span>File ▾</span>, submenu: fileSubmenu },
+  { key: 'edit', content: <span>Edit ▾</span>, submenu: editSubmenu },
+  { key: 'format', content: <span>Format ▾</span>, submenu: formatSubmenu },
+  { key: 'help', content: 'Help' },
+]
+
+const MenuExampleWithSubmenu = () => (
+  <Menu defaultActiveIndex={0} items={items} variables={menuWidthVariable} />
+)
```
Comment (mnajdova) [inline docs/src/examples/components/Menu/Types/MenuExampleWithSubmenu.shorthand.tsx:36]: Please try to add some height on the element as the content is not visible (try to wrap the Menu in some wrapper with some bigger height). 
image

Comment (mnajdova) [reply docs/src/examples/components/Menu/Types/MenuExampleWithSubmenu.shorthand.tsx:36]: Also, I would expect to be able to expand only one submenu at a time. I don't think the scenario above (having three different dropdowns should work (try closing the submenu if there is click outside of it).

File: src/themes/teams/components/Menu/menuItemStyles.ts [hunk 3]
```diff
@@ -238,6 +239,11 @@ const menuItemStyles = {
         }),
     }
   },
+  submenu: {
+    position: 'absolute',
+    borderTopLeftRadius: 0,
+    borderTopRightRadius: 0,
+  },
```
Comment (mnajdova) [inline src/themes/teams/components/Menu/menuItemStyles.ts:246]: Totally support @notandrew comment, the type should affect the submenu as well.

**Comment by levithomason** [comment]
@priyankar205 and @gopalgoel19 please review this and collaborate.  It seems there is some overlapping work here.

**Comment by sophieH29** [comment]
Closing as it doesn't support our immediate needs right now

**Pull request closed without merging**